utils/build-script-impl

```diff
@@ -990,20 +990,22 @@ fi
 if [[ ! "${SKIP_BUILD_LLBUILD}" ]] ; then
      PRODUCTS=("${PRODUCTS[@]}" llbuild)
 fi
-if [[ ! "${SKIP_BUILD_SWIFTPM}" ]] ; then
-     PRODUCTS=("${PRODUCTS[@]}" swiftpm)
-fi
 if [[ ! "${SKIP_BUILD_LIBDISPATCH}" ]] ; then
      PRODUCTS=("${PRODUCTS[@]}" libdispatch)
 fi
-# XCTest has a dependency on Foundation, so Foundation must be added to the
-# list of build products first.
+# SwiftPM and XCTest are dependent on Foundation, so Foundation must be
+# added to the list of build products first.
 if [[ ! "${SKIP_BUILD_FOUNDATION}" ]] ; then
      PRODUCTS=("${PRODUCTS[@]}" foundation)
 fi
+# SwiftPM is dependent on XCTest, so XCTest must be added to the list of
```

Comment:
It doesn't currently, but it does actually use XCTest today (whereas the converse is not true), and I can imagine at some point it might be desirable to "just build".

I think we are going to ultimately have to solve the circular build dependency another way, but would rather start with this order since SwiftPM does use XCTest.